tests/unit/components/playback/controls/SettingsFormButtons.spec.js

```diff
@@ -0,0 +1,652 @@
+  });
+
+  beforeEach(async () => {
+    store = await NuxtStore.createStore();
+  });
+
+  afterEach(() => {
+    wrapper.destroy();
+  });
+
+  test("Given the SettingsForm has a valid customer account set as 'Customer account -1, When the value is set as focus Customer ID, Then a valid rules are executed and relevant buttons are enabled rest are greyed", async () => {
+    store.commit("settings/set_customer_account_ids", array_of_customerids);
+
+    wrapper = mount(ComponentToTest, {
+      store,
+      localVue,
+    });
+
+    wrapper
+      .find("#input-dropdown-widget-cust-")
+      .setValue("Customer account -1");
+
+    const add_customer_btn = wrapper.find(
+      ".span__settingsform-customer-add-btn_txt"
+    );
+    const edit_customer_btn = wrapper.find(
+      ".span__settingsform-customer-edit-btn-txt"
+    );
+    const add_user_btn = wrapper.find(".span__settingsform_user-input-editor");
+    const edit_user_btn = wrapper.find(
+      ".span__settingsform-user-input-edit-btn-txt"
+    );
+
+    await wrapper.vm.$nextTick(); // wait for update
+
+    expect(add_customer_btn.isVisible()).toBe(true);
+    expect(edit_customer_btn.isVisible()).toBe(true);
+    expect(add_user_btn.isVisible()).toBe(true);
+    expect(edit_user_btn.isVisible()).toBe(false);
+  });
+  test("Given the SettingsForm has a valid customer account set as 'Customer account -1'and 'User account -1', When the value is set as focus Customer ID and User ID, Then a valid rules are executed and all buttons are enabled none of the buttons are greyed", async () => {
+    store.commit("settings/set_customer_account_ids", array_of_customerids);
+
+    wrapper = mount(ComponentToTest, {
+      store,
+      localVue,
+    });
+
+    await wrapper
+      .find("#input-dropdown-widget-cust-")
+      .setValue("Customer account -1");
+    await wrapper.vm.$nextTick(); // wait for update
+
+    await wrapper
+      .find("#input-dropdown-widget-user-")
+      .setValue("User account -1");
+    await wrapper.vm.$nextTick(); // wait for update
+
+    const add_customer_btn = wrapper.find(
+      ".span__settingsform-customer-add-btn_txt"
+    );
+    const edit_customer_btn = wrapper.find(
+      ".span__settingsform-customer-edit-btn-txt"
+    );
+    const add_user_btn = wrapper.find(".span__settingsform_user-input-editor");
+    const edit_user_btn = wrapper.find(
+      ".span__settingsform-user-input-edit-btn-txt"
+    );
+
+    expect(add_customer_btn.isVisible()).toBe(true);
+    expect(edit_customer_btn.isVisible()).toBe(true);
+    expect(add_user_btn.isVisible()).toBe(true);
+    expect(edit_user_btn.isVisible()).toBe(true);
+  });
+
+  test("Given the SettingsForm and neither customer nor user 'ID'  set, When the value is set as focus Customer ID/User ID is <empty>, Then a valid rules are executed and only 'Add Customer ID' is in focus the rest are greyed", async () => {
+    store.commit("settings/set_customer_account_ids", array_of_customerids);
+
```

Comment:
For now there is a common label name 'Edit ID', duplicated for both customer/user so for titles its modified as below:- "the rest are greyed" is removed.
a) 'Add New Customer ID'
b) 'Edit ID'(of customer)
c) 'Add New User ID'
d) 'Edit ID'(of user) 

Hope this will help.

Yes Vuex data is essential as initial load looks up the settings store and determines
the Customer ID <text> and User ID <text> and  buttons enable and disable starts from that point onwards.  Having said the above assume an interaction may  lead to "Delete ID" on Customer ID which is focus due to customer_index/user_index value set needs to reset in Vuex, then user has to set an alternative as current Customer ID and User ID, so in such scenario the Button enabled and disabled  validation are bounded to Vuex store.  So that once all changes are done the user perform "SaveChanges".

Sorry we can't remove the Vuex store data in the testcases it becomes very complex to verify the Buttons enable and disable.  I don't know how we can inject list of customers into a an array called customer_account_ids, customer_index and user_index, which is directly bounded to the Vuex store due to a  requirement that states, on load the value present in the Vuex of settings has to be shown on Customer ID and User ID if specified in the settings store.  Is there an alternative way to set the values without storing in Vuex  as the are available via computed Api's.